src/Visitor/CoreVisitor.php

```diff
@@ -46,7 +47,24 @@ public function visit (MenuItem $item) : void
         // replace target with URL
         if ($target instanceof LazyRoute)
         {
-            $item->setTarget($target->generate($this->urlGenerator));
+            // store the previous route in an extra attribute
+            $item->setExtra("_route", $target->getRoute());
+
+            try
+            {
+                $item->setTarget($target->generate($this->urlGenerator));
+            }
+            catch (MissingMandatoryParametersException $exception)
+            {
+                // ignore exception if no parameters were given
+                // otherwise -> rethrow
+                if (!empty($target->getParameters()))
```

Comment:
Shouldn't we always re-throw it? Otherwise I might not even notice that some URLs of my large menu might not be working if I simply forgot to pass in parameters.

Reply:
Maybe we could change the behaviour based on the current environment, just like with our AssetsBundle: Make it throw in dev but silently bail out in prod. Would that be a better solution?

Reply:
With the route tree bundle you have around 100 routes that have no default parameters set.
Except you want to reintroduce the huge chunk of code that deals with fake defaults... which I would avoid.

Reply:
You can always just implement a visitor that iterates through all routes and validates that they are compilable, that is rather easy.

Reply:
I just changed the whole implementation, so that only the `LazyRoute`s are resolved, that are actually visible. Then we can actually log these errors now, as they won't flood the log with insignificant errors.

Reply:
Perfect, thank you!